docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+
+
+Implement default parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For spectral models Gammapy has defined default parameters, which simplifes the
+interactive usage of the models. From this experience we propose to introduce
+default parameters for `SpatialModel` as well as `SourceModel`.
+
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+    from gammapy.cube.models import SourceModels
+
+    shell = Shell()
+    print(shell)
+
+
+Implement evaluation region specifications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For efficient evaluation of model components, we propose that `SpatialModel`
+classes declare a circular region `.evaluation_radius`, they are evaluated on.
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+
+    shell = Shell()
+    print(shell.evaluation_radius)
+
+The `.evaluation_radius` should be a property of the model, that is computed
+depending of the model parameter, that determines the spatial extension of the
+model. The handling of `PointSource` and additional evaluation margins, which depend
+on the pixel size, is done in the step, where the model component is evaluated.
+
+
+Expose model parameters as attributes
+-------------------------------------
+To simplify the interactive handling of model instances, model parameters should
+be exposed as attribues on the model, so that the following works:
+
+.. code-block::
+
+    from gammapy.spectrum.models import Powerlaw
+
+    pwl = Powerlaw()
+
+    pwl.amplitude.value = "1e-12 cm-2 s-1 TeV-1"
+
+    # instead of
+
+    pwl.parameters["amplitude"].value = "1e-12 cm-2 s-1 TeV-1"
+
+The attributes should be auto-genrated in the `Model` base class, based on the
+list of parameters defined by the model.
+
+Introduce sub-module for models
+-------------------------------
+
+We propose to introduce a `gammapy.models` or `gammapy.modeling` sub-module, which
+bundles all the model classes in single namespace.
+
+.. code-block::
+
+    from gammapy.models import Powerlaw, SpatialGaussian, SourceModel
+
+    # or consistently with astropy
+
+    from gammapy.modeling import Powerlaw, SpatialGaussian, SourceModel
+
+
+List of Pull Requests
+=====================
+
+This is a proposal for a list of pull requests implementing the proposed changes,
+order by priority:
+
+- Implement a `BackgroundModel` base class and a `BackgroundIRFModel` class. Use the
+  `BackgroundIRFModel` object in the current `MapEvaluator`. Join the background
+  parameters with the source model parameters and expose the joined parameter list
+  at the `MapEvaluator` object, so that it can be used be used in current `MapFit`.
+
+- Reimplement the "+" operator for `SkyModel` using the `SkyModels` class.
+  Remove the `CompoundSkyModel` class.
+
+- Implement support for model component names in `SkyModel`. Implement `SkyModels.__getitem__`
+  to allow for access of model components by name.
+
+- Fix existing XML I/O of `SkyModels` and add serialization of the model component
```

Comment:
The model serialisation is a big and pretty much independent task. Maybe put it at the end?
E.g. currently you have `evaluation_radius` or sky coord frame below model serialisation in terms of priority.
Shouldn't those be higher priority?